app/assets/javascripts/discourse/app/components/topic-timeline/scroll-area.js

```diff
@@ -0,0 +1,231 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { bind } from "discourse-common/utils/decorators";
+import { tracked } from "@glimmer/tracking";
+import discourseLater from "discourse-common/lib/later";
+import { action } from "@ember/object";
+
+export const SCROLLER_HEIGHT = 50;
+const MIN_SCROLLAREA_HEIGHT = 170;
+const MAX_SCROLLAREA_HEIGHT = 300;
+const LAST_READ_HEIGHT = 20;
+
+export default class TopicTimelineScrollArea extends GlimmerComponent {
+  @tracked showButton = false;
+  @tracked scrollPosition;
+  @tracked current;
+  @tracked percentage = this._percentFor(
+    this.args.topic,
+    this.args.enteredIndex + 1
+  );
+  @tracked total;
+  @tracked date;
+  @tracked lastReadPercentage = null;
+  @tracked position;
+  @tracked lastReadTop = Math.round(
+    this.lastReadPercentage * scrollareaHeight()
+  );
+
+  style = `height: ${scrollareaHeight()}px`;
+  before = this.scrollareaRemaining() * this.percentage;
+  after = scrollareaHeight() - this.before - SCROLLER_HEIGHT;
```

Comment:
These all need to be a getters to make sure they're re-evalated when the args change